app/models/user.rb

```diff
@@ -90,4 +99,25 @@ def match?(user)
   def minus_friend(user)
     friends - [User.find(user)]
   end
+
+  def facebook
+    # You need to implement the method below in your model (e.g. app/models/user.rb)
+    @user = User.from_omniauth(request.env["omniauth.auth"])
+
+    if @user.persisted?
+      sign_in_and_redirect @user, :event => :authentication #this will throw if @user is not activated
```

Comment:
Line is too long. [102/80]<br>Use the new Ruby 1.9 hash syntax.<br>Missing space after #.